governator-core/src/main/java/com/netflix/governator/event/ApplicationEventModule.java

```diff
@@ -0,0 +1,115 @@
+        public GuavaApplicationEventPublisher(EventBus eventBus) {
+            this.eventBus = eventBus;  
+        }
+
+        @Override
+        public void publishEvent(ApplicationEvent event) {
+            this.eventBus.post(event);
+        }
+        
+    }
+    
+    @Singleton
+    private static class ApplicationEventSubscribingTypeListener implements TypeListener {
+       
+        private EventBus eventBus;
+        
+        public ApplicationEventSubscribingTypeListener(EventBus eventBus) {
+            this.eventBus = eventBus;
+        }
+
+        @Override
+        public <I> void hear(TypeLiteral<I> type, TypeEncounter<I> encounter) {
+            Class<?> clazz = type.getRawType();
+            while (clazz != null && !Collection.class.isAssignableFrom(clazz) && !clazz.isArray()) {
+              for (final Method method : clazz.getDeclaredMethods()) {
+                  if(method.isAnnotationPresent(EventListener.class)) {
+                      if(method.getReturnType().equals(Void.TYPE)
+                              && method.getParameterTypes().length == 1 
+                              && ApplicationEvent.class.isAssignableFrom(method.getParameterTypes()[0])) {
+                            encounter.register(new InjectionListener<Object>() {
+                                @Override
+                                public void afterInjection(Object injectee) {
+                                    GuavaSubscriberProxy proxy = new GuavaSubscriberProxy(injectee, method);
+                                    eventBus.register(proxy);
+                                }
+                            });
+                      }
+                      else
+                      {
+                          LOG.warn("@EventListener {}.{} skipped. Methods must be public, void, and accept exactly"
+                                  + " one argument extending com.netflix.governator.event.ApplicationEvent.", clazz.getName(), method.getName());
+                      }
+                  }
+              }
+              clazz = clazz.getSuperclass();   
+          }
+        }
+    }
+    
+    private static class GuavaSubscriberProxy { 
+        
+        private final Object handlerInstance;
+        private final Method handlerMethod;
+        
+        public GuavaSubscriberProxy(Object handlerInstance, Method handlerMethod) {
+            this.handlerInstance = handlerInstance;
+            this.handlerMethod = handlerMethod;
+        }
+        
+        @Subscribe
+        public void invokeEventHandler(ApplicationEvent event) throws IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+            if(!handlerMethod.isAccessible()) {
+                handlerMethod.setAccessible(true);
+            }    
+            handlerMethod.invoke(handlerInstance, event);           
+        }
+    }
+
+    @Override
+    protected void configure() {
+        bind(EventBus.class).toInstance(eventBus);
```

Comment:
Instead of binding to an instance here, you can create the eventBus instance via an @ Provides and @ Named evenBus instance for injection into GuavaApplicationEventPublisher.